client/src/components/WorkflowInvocationState/WorkflowInvocationState.vue

```diff
@@ -108,18 +108,22 @@ const invocationSchedulingTerminal = computed(() => {
     );
 });
 const jobStatesTerminal = computed(() => {
+    // If the job states summary is null, we haven't fetched it yet
+    if (jobStatesSummary.value === null) {
+        return false;
+    }
+
     if (invocationSchedulingTerminal.value && jobCount.value === 0) {
-        // no jobs for this invocation (think subworkflow or just inputs)
+        // no jobs for this invocation (think it has just subworkflows/inputs)
         return true;
```

Comment:
This isn't fixing it entirely unfortunately.

We still get a case where **sometimes**, we fetch the job states summary just in time, that it is:
```
{ "id": "id_here", "states": {}, "populated_state": "new", "model": "WorkflowInvocation" } 
```
instead of
```
{ "id": "id_here", "states": { "queued": 2 }, "populated_state": "new", "model": "WorkflowInvocation" } 
```
And before the summary can get non terminal states, like so ^, `invocationSchedulingTerminal` becomes `true`, and so, the `jobCount` is considered 0, and we stop fetching the job summaries; even though we really need to since we see 2 steps with 2 expected jobs, like:

![image](https://github.com/user-attachments/assets/8b653a33-f8ec-4c0c-81b4-3e2fc6701d1b)

Reply:
So this next commit https://github.com/galaxyproject/galaxy/pull/20010/commits/642ad39a5dde2c4ee831e07e5daf179d564ca981 should fix it if I got the concept right:

We see the `populated_state` being set in `summarize_invocation_jobs` here https://github.com/galaxyproject/galaxy/blob/4a0cca07d97d7bbd22eb39ddab38d4d4354aa4fa/lib/galaxy/managers/jobs.py#L957-L1000, `new` seems to be a non-terminal `populated_state`, meaning this summary will have updates which will eventually produce some state counts.

To fix the bug of never getting a state count for the invocation view before the scheduling state becomes terminal, we now wait for the state to be something else than `new`, and only then look for other reasons to stop polling.